src/answers-headless.ts

```diff
@@ -261,7 +271,12 @@ export default class AnswersHeadless {
    *
    * @returns A Promise of a {@link UniversalSearchResponse} from the Answers API
    */
-  async executeUniversalQuery(): Promise<UniversalSearchResponse> {
+  async executeUniversalQuery(): Promise<UniversalSearchResponse | undefined> {
```

Comment:
I know that AlternativeVertical component in vertical page uses executeUniversalQuery to get results from all verticals, so it would have to switch between searchTypes from Vertical => Universal (execute search) => Vertical right? This would require us to pass a field, verticalKey, around for any component that requires this crossover logic. But, I think this is the only spot that cross between universal and vertical searches so this should be ok? just double checking

Reply:
AlternativeVerticals uses the `alternativeVerticals` and the `allResultsForVertical` data which are available from a vertical query. I don't believe it uses `executeUniversalQuery`

Reply:
oh I saw in the code where it calls `renderUniversalDetails` which calls `actions.executeUniversalQuery`. Is that something else? Or maybe it's not updated?

Reply:
Oh, you are right. It does fire off a universal search, how that was because we didn't have a universal results component or the page linking figured out at that time. I think we should update that component so that clicking on the links works by going through the react router, and the page navigated to should fire off the search. So universal search inside Alternative verticals shouldn't be an issue after those updates